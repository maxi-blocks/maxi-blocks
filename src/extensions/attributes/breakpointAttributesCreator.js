const breakpoints = ['g', 'xxl', 'xl', 'l', 'm', 's', 'xs'];

const breakpointAttributesCreator = ({
	obj,
	noBreakpointAttr = [],
	diffValAttr = {},
}) => {
	const response = {};
	const diffValAttrKeys = Object.keys(diffValAttr);

	Object.entries(obj).forEach(([key, val]) => {
		if (noBreakpointAttr?.includes(key)) {
			response[key] = val;

			return;
		}

		breakpoints.forEach(breakpoint => {
			const newKey = `${key}-${breakpoint}`;
			const newVal = {
				...val,
				longLabel: `${val.longLabel}-${breakpoint}`,
			};

			if (diffValAttrKeys.includes(newKey))
				newVal.default = diffValAttr[newKey];
			else if (breakpoint !== 'g') delete newVal.default;

			response[newKey] = newVal;
		});
	});

	return response;
};

export default breakpointAttributesCreator;
