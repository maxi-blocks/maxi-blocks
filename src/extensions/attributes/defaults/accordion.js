const accordion = {
	_acl: {
		type: 'string',
		default: 'simple',
		longLabel: 'accordionLayout',
	},
	_apc: {
		type: 'boolean',
		default: true,
		longLabel: 'autoPaneClose',
	},
	_ico: {
		type: 'boolean',
		default: true,
		longLabel: 'isCollapsible',
	},
	_ad: {
		type: 'number',
		default: 0.3,
		longLabel: 'animationDuration',
	},
};

export default accordion;
