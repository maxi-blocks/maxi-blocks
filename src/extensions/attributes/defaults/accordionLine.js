import prefixAttributesCreator from '../prefixAttributesCreator';
import divider from './divider';
import dividerHover from './dividerHover';

const accordionLineAttributes = {
	'li.sh': {
		type: 'boolean',
		default: false,
		longLabel: 'line-status-hover',
	},
	'li.sa': {
		type: 'boolean',
		default: false,
		longLabel: 'line-status-active',
	},
};

const rawAccordionLine = {
	...divider,
	'di-bo_pc-g': {
		type: 'number',
		default: 3,
		longLabel: 'divider-border-palette-color-general',
	},
	'di_w-g': {
		type: 'number',
		default: 100,
		longLabel: 'divider-width-general',
	},
	...dividerHover,
	'di_w-g.h': {
		type: 'number',
		default: 100,
		longLabel: 'divider-width-general-hover',
	},
	...accordionLineAttributes,
	...prefixAttributesCreator({
		obj: divider,
		prefix: 'a-',
		longPrefix: 'active-',
		diffValAttr: {
			'a-db_pc-g': 3, // active-divider-border-palette-color-g
			'a-di_w-g': 100, // active-divider-width-g
		},
	}),
};

const accordionLine = {
	...prefixAttributesCreator({
		obj: rawAccordionLine,
		prefix: 'he-',
		longPrefix: 'header-',
	}),
	...prefixAttributesCreator({
		obj: rawAccordionLine,
		prefix: 'c-',
		longPrefix: 'content-',
	}),
};

export default accordionLine;
