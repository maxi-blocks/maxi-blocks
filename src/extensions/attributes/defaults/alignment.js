import breakpointAttributesCreator from '../breakpointAttributesCreator';
import attributesShorter from '../dictionary/attributesShorter';

const alignment = breakpointAttributesCreator({
	obj: {
		alignment: {
			type: 'string',
			default: 'center',
		},
	},
});

export default attributesShorter(alignment, 'alignment');
