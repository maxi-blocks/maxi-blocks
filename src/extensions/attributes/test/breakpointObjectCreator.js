import { rawBackgroundImage } from '../defaults/background';
import breakpointAttributesCreator from '../breakpointAttributesCreator';

describe('breakpointAttributesCreator', () => {
	it('Returns an image bg object', () => {
		const result = breakpointAttributesCreator({
			obj: rawBackgroundImage,
			noBreakpointAttr: ['bi_mu', 'bi_mi', 'bi_pal', 'bi_pas'],
		});

		expect(result).toMatchSnapshot();
	});
});
