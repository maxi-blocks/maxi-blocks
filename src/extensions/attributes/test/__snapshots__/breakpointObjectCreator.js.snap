// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakpointAttributesCreator Returns an image bg object 1`] = `
Object {
  "bi_at-g": Object {
    "default": "scroll",
    "longLabel": "background-image-attachment-g",
    "type": "string",
  },
  "bi_at-l": Object {
    "longLabel": "background-image-attachment-l",
    "type": "string",
  },
  "bi_at-m": Object {
    "longLabel": "background-image-attachment-m",
    "type": "string",
  },
  "bi_at-s": Object {
    "longLabel": "background-image-attachment-s",
    "type": "string",
  },
  "bi_at-xl": Object {
    "longLabel": "background-image-attachment-xl",
    "type": "string",
  },
  "bi_at-xs": Object {
    "longLabel": "background-image-attachment-xs",
    "type": "string",
  },
  "bi_at-xxl": Object {
    "longLabel": "background-image-attachment-xxl",
    "type": "string",
  },
  "bi_clp-g": Object {
    "default": "border-box",
    "longLabel": "background-image-clip-g",
    "type": "string",
  },
  "bi_clp-l": Object {
    "longLabel": "background-image-clip-l",
    "type": "string",
  },
  "bi_clp-m": Object {
    "longLabel": "background-image-clip-m",
    "type": "string",
  },
  "bi_clp-s": Object {
    "longLabel": "background-image-clip-s",
    "type": "string",
  },
  "bi_clp-xl": Object {
    "longLabel": "background-image-clip-xl",
    "type": "string",
  },
  "bi_clp-xs": Object {
    "longLabel": "background-image-clip-xs",
    "type": "string",
  },
  "bi_clp-xxl": Object {
    "longLabel": "background-image-clip-xxl",
    "type": "string",
  },
  "bi_co-g": Object {
    "longLabel": "background-image-crop-options-g",
    "type": "object",
  },
  "bi_co-l": Object {
    "longLabel": "background-image-crop-options-l",
    "type": "object",
  },
  "bi_co-m": Object {
    "longLabel": "background-image-crop-options-m",
    "type": "object",
  },
  "bi_co-s": Object {
    "longLabel": "background-image-crop-options-s",
    "type": "object",
  },
  "bi_co-xl": Object {
    "longLabel": "background-image-crop-options-xl",
    "type": "object",
  },
  "bi_co-xs": Object {
    "longLabel": "background-image-crop-options-xs",
    "type": "object",
  },
  "bi_co-xxl": Object {
    "longLabel": "background-image-crop-options-xxl",
    "type": "object",
  },
  "bi_cp-g": Object {
    "default": "none",
    "longLabel": "background-image-clip-path-g",
    "type": "string",
  },
  "bi_cp-l": Object {
    "longLabel": "background-image-clip-path-l",
    "type": "string",
  },
  "bi_cp-m": Object {
    "longLabel": "background-image-clip-path-m",
    "type": "string",
  },
  "bi_cp-s": Object {
    "longLabel": "background-image-clip-path-s",
    "type": "string",
  },
  "bi_cp-xl": Object {
    "longLabel": "background-image-clip-path-xl",
    "type": "string",
  },
  "bi_cp-xs": Object {
    "longLabel": "background-image-clip-path-xs",
    "type": "string",
  },
  "bi_cp-xxl": Object {
    "longLabel": "background-image-clip-path-xxl",
    "type": "string",
  },
  "bi_cp.s-g": Object {
    "default": false,
    "longLabel": "background-image-clip-path-status-g",
    "type": "boolean",
  },
  "bi_cp.s-l": Object {
    "longLabel": "background-image-clip-path-status-l",
    "type": "boolean",
  },
  "bi_cp.s-m": Object {
    "longLabel": "background-image-clip-path-status-m",
    "type": "boolean",
  },
  "bi_cp.s-s": Object {
    "longLabel": "background-image-clip-path-status-s",
    "type": "boolean",
  },
  "bi_cp.s-xl": Object {
    "longLabel": "background-image-clip-path-status-xl",
    "type": "boolean",
  },
  "bi_cp.s-xs": Object {
    "longLabel": "background-image-clip-path-status-xs",
    "type": "boolean",
  },
  "bi_cp.s-xxl": Object {
    "longLabel": "background-image-clip-path-status-xxl",
    "type": "boolean",
  },
  "bi_h-g": Object {
    "default": 100,
    "longLabel": "background-image-height-g",
    "type": "number",
  },
  "bi_h-l": Object {
    "longLabel": "background-image-height-l",
    "type": "number",
  },
  "bi_h-m": Object {
    "longLabel": "background-image-height-m",
    "type": "number",
  },
  "bi_h-s": Object {
    "longLabel": "background-image-height-s",
    "type": "number",
  },
  "bi_h-xl": Object {
    "longLabel": "background-image-height-xl",
    "type": "number",
  },
  "bi_h-xs": Object {
    "longLabel": "background-image-height-xs",
    "type": "number",
  },
  "bi_h-xxl": Object {
    "longLabel": "background-image-height-xxl",
    "type": "number",
  },
  "bi_h.u-g": Object {
    "default": "%",
    "longLabel": "background-image-height-unit-g",
    "type": "string",
  },
  "bi_h.u-l": Object {
    "longLabel": "background-image-height-unit-l",
    "type": "string",
  },
  "bi_h.u-m": Object {
    "longLabel": "background-image-height-unit-m",
    "type": "string",
  },
  "bi_h.u-s": Object {
    "longLabel": "background-image-height-unit-s",
    "type": "string",
  },
  "bi_h.u-xl": Object {
    "longLabel": "background-image-height-unit-xl",
    "type": "string",
  },
  "bi_h.u-xs": Object {
    "longLabel": "background-image-height-unit-xs",
    "type": "string",
  },
  "bi_h.u-xxl": Object {
    "longLabel": "background-image-height-unit-xxl",
    "type": "string",
  },
  "bi_mi": Object {
    "longLabel": "background-image-mediaID",
    "type": "number",
  },
  "bi_mu": Object {
    "longLabel": "background-image-mediaURL",
    "type": "string",
  },
  "bi_o-g": Object {
    "default": 1,
    "longLabel": "background-image-opacity-g",
    "type": "number",
  },
  "bi_o-l": Object {
    "longLabel": "background-image-opacity-l",
    "type": "number",
  },
  "bi_o-m": Object {
    "longLabel": "background-image-opacity-m",
    "type": "number",
  },
  "bi_o-s": Object {
    "longLabel": "background-image-opacity-s",
    "type": "number",
  },
  "bi_o-xl": Object {
    "longLabel": "background-image-opacity-xl",
    "type": "number",
  },
  "bi_o-xs": Object {
    "longLabel": "background-image-opacity-xs",
    "type": "number",
  },
  "bi_o-xxl": Object {
    "longLabel": "background-image-opacity-xxl",
    "type": "number",
  },
  "bi_ori-g": Object {
    "default": "padding-box",
    "longLabel": "background-image-origin-g",
    "type": "string",
  },
  "bi_ori-l": Object {
    "longLabel": "background-image-origin-l",
    "type": "string",
  },
  "bi_ori-m": Object {
    "longLabel": "background-image-origin-m",
    "type": "string",
  },
  "bi_ori-s": Object {
    "longLabel": "background-image-origin-s",
    "type": "string",
  },
  "bi_ori-xl": Object {
    "longLabel": "background-image-origin-xl",
    "type": "string",
  },
  "bi_ori-xs": Object {
    "longLabel": "background-image-origin-xs",
    "type": "string",
  },
  "bi_ori-xxl": Object {
    "longLabel": "background-image-origin-xxl",
    "type": "string",
  },
  "bi_pa.s-g": Object {
    "default": false,
    "longLabel": "background-image-parallax-status-g",
    "type": "boolean",
  },
  "bi_pa.s-l": Object {
    "longLabel": "background-image-parallax-status-l",
    "type": "boolean",
  },
  "bi_pa.s-m": Object {
    "longLabel": "background-image-parallax-status-m",
    "type": "boolean",
  },
  "bi_pa.s-s": Object {
    "longLabel": "background-image-parallax-status-s",
    "type": "boolean",
  },
  "bi_pa.s-xl": Object {
    "longLabel": "background-image-parallax-status-xl",
    "type": "boolean",
  },
  "bi_pa.s-xs": Object {
    "longLabel": "background-image-parallax-status-xs",
    "type": "boolean",
  },
  "bi_pa.s-xxl": Object {
    "longLabel": "background-image-parallax-status-xxl",
    "type": "boolean",
  },
  "bi_pal": Object {
    "longLabel": "background-image-parallax-alt",
    "type": "string",
  },
  "bi_pas": Object {
    "longLabel": "background-image-parallax-alt-selector",
    "type": "string",
  },
  "bi_pd-g": Object {
    "default": "down",
    "longLabel": "background-image-parallax-direction-g",
    "type": "string",
  },
  "bi_pd-l": Object {
    "longLabel": "background-image-parallax-direction-l",
    "type": "string",
  },
  "bi_pd-m": Object {
    "longLabel": "background-image-parallax-direction-m",
    "type": "string",
  },
  "bi_pd-s": Object {
    "longLabel": "background-image-parallax-direction-s",
    "type": "string",
  },
  "bi_pd-xl": Object {
    "longLabel": "background-image-parallax-direction-xl",
    "type": "string",
  },
  "bi_pd-xs": Object {
    "longLabel": "background-image-parallax-direction-xs",
    "type": "string",
  },
  "bi_pd-xxl": Object {
    "longLabel": "background-image-parallax-direction-xxl",
    "type": "string",
  },
  "bi_pos-g": Object {
    "default": "center center",
    "longLabel": "background-image-position-g",
    "type": "string",
  },
  "bi_pos-l": Object {
    "longLabel": "background-image-position-l",
    "type": "string",
  },
  "bi_pos-m": Object {
    "longLabel": "background-image-position-m",
    "type": "string",
  },
  "bi_pos-s": Object {
    "longLabel": "background-image-position-s",
    "type": "string",
  },
  "bi_pos-xl": Object {
    "longLabel": "background-image-position-xl",
    "type": "string",
  },
  "bi_pos-xs": Object {
    "longLabel": "background-image-position-xs",
    "type": "string",
  },
  "bi_pos-xxl": Object {
    "longLabel": "background-image-position-xxl",
    "type": "string",
  },
  "bi_pos_h-g": Object {
    "default": 0,
    "longLabel": "background-image-position-height-g",
    "type": "number",
  },
  "bi_pos_h-l": Object {
    "longLabel": "background-image-position-height-l",
    "type": "number",
  },
  "bi_pos_h-m": Object {
    "longLabel": "background-image-position-height-m",
    "type": "number",
  },
  "bi_pos_h-s": Object {
    "longLabel": "background-image-position-height-s",
    "type": "number",
  },
  "bi_pos_h-xl": Object {
    "longLabel": "background-image-position-height-xl",
    "type": "number",
  },
  "bi_pos_h-xs": Object {
    "longLabel": "background-image-position-height-xs",
    "type": "number",
  },
  "bi_pos_h-xxl": Object {
    "longLabel": "background-image-position-height-xxl",
    "type": "number",
  },
  "bi_pos_h.u-g": Object {
    "default": "%",
    "longLabel": "background-image-position-height-unit-g",
    "type": "string",
  },
  "bi_pos_h.u-l": Object {
    "longLabel": "background-image-position-height-unit-l",
    "type": "string",
  },
  "bi_pos_h.u-m": Object {
    "longLabel": "background-image-position-height-unit-m",
    "type": "string",
  },
  "bi_pos_h.u-s": Object {
    "longLabel": "background-image-position-height-unit-s",
    "type": "string",
  },
  "bi_pos_h.u-xl": Object {
    "longLabel": "background-image-position-height-unit-xl",
    "type": "string",
  },
  "bi_pos_h.u-xs": Object {
    "longLabel": "background-image-position-height-unit-xs",
    "type": "string",
  },
  "bi_pos_h.u-xxl": Object {
    "longLabel": "background-image-position-height-unit-xxl",
    "type": "string",
  },
  "bi_pos_w-g": Object {
    "default": 0,
    "longLabel": "background-image-position-width-g",
    "type": "number",
  },
  "bi_pos_w-l": Object {
    "longLabel": "background-image-position-width-l",
    "type": "number",
  },
  "bi_pos_w-m": Object {
    "longLabel": "background-image-position-width-m",
    "type": "number",
  },
  "bi_pos_w-s": Object {
    "longLabel": "background-image-position-width-s",
    "type": "number",
  },
  "bi_pos_w-xl": Object {
    "longLabel": "background-image-position-width-xl",
    "type": "number",
  },
  "bi_pos_w-xs": Object {
    "longLabel": "background-image-position-width-xs",
    "type": "number",
  },
  "bi_pos_w-xxl": Object {
    "longLabel": "background-image-position-width-xxl",
    "type": "number",
  },
  "bi_pos_w.u-g": Object {
    "default": "%",
    "longLabel": "background-image-position-width-unit-g",
    "type": "string",
  },
  "bi_pos_w.u-l": Object {
    "longLabel": "background-image-position-width-unit-l",
    "type": "string",
  },
  "bi_pos_w.u-m": Object {
    "longLabel": "background-image-position-width-unit-m",
    "type": "string",
  },
  "bi_pos_w.u-s": Object {
    "longLabel": "background-image-position-width-unit-s",
    "type": "string",
  },
  "bi_pos_w.u-xl": Object {
    "longLabel": "background-image-position-width-unit-xl",
    "type": "string",
  },
  "bi_pos_w.u-xs": Object {
    "longLabel": "background-image-position-width-unit-xs",
    "type": "string",
  },
  "bi_pos_w.u-xxl": Object {
    "longLabel": "background-image-position-width-unit-xxl",
    "type": "string",
  },
  "bi_psp-g": Object {
    "default": 4,
    "longLabel": "background-image-parallax-speed-g",
    "type": "number",
  },
  "bi_psp-l": Object {
    "longLabel": "background-image-parallax-speed-l",
    "type": "number",
  },
  "bi_psp-m": Object {
    "longLabel": "background-image-parallax-speed-m",
    "type": "number",
  },
  "bi_psp-s": Object {
    "longLabel": "background-image-parallax-speed-s",
    "type": "number",
  },
  "bi_psp-xl": Object {
    "longLabel": "background-image-parallax-speed-xl",
    "type": "number",
  },
  "bi_psp-xs": Object {
    "longLabel": "background-image-parallax-speed-xs",
    "type": "number",
  },
  "bi_psp-xxl": Object {
    "longLabel": "background-image-parallax-speed-xxl",
    "type": "number",
  },
  "bi_re-g": Object {
    "default": "no-repeat",
    "longLabel": "background-image-repeat-g",
    "type": "string",
  },
  "bi_re-l": Object {
    "longLabel": "background-image-repeat-l",
    "type": "string",
  },
  "bi_re-m": Object {
    "longLabel": "background-image-repeat-m",
    "type": "string",
  },
  "bi_re-s": Object {
    "longLabel": "background-image-repeat-s",
    "type": "string",
  },
  "bi_re-xl": Object {
    "longLabel": "background-image-repeat-xl",
    "type": "string",
  },
  "bi_re-xs": Object {
    "longLabel": "background-image-repeat-xs",
    "type": "string",
  },
  "bi_re-xxl": Object {
    "longLabel": "background-image-repeat-xxl",
    "type": "string",
  },
  "bi_si-g": Object {
    "default": "auto",
    "longLabel": "background-image-size-g",
    "type": "string",
  },
  "bi_si-l": Object {
    "longLabel": "background-image-size-l",
    "type": "string",
  },
  "bi_si-m": Object {
    "longLabel": "background-image-size-m",
    "type": "string",
  },
  "bi_si-s": Object {
    "longLabel": "background-image-size-s",
    "type": "string",
  },
  "bi_si-xl": Object {
    "longLabel": "background-image-size-xl",
    "type": "string",
  },
  "bi_si-xs": Object {
    "longLabel": "background-image-size-xs",
    "type": "string",
  },
  "bi_si-xxl": Object {
    "longLabel": "background-image-size-xxl",
    "type": "string",
  },
  "bi_w-g": Object {
    "default": 100,
    "longLabel": "background-image-width-g",
    "type": "number",
  },
  "bi_w-l": Object {
    "longLabel": "background-image-width-l",
    "type": "number",
  },
  "bi_w-m": Object {
    "longLabel": "background-image-width-m",
    "type": "number",
  },
  "bi_w-s": Object {
    "longLabel": "background-image-width-s",
    "type": "number",
  },
  "bi_w-xl": Object {
    "longLabel": "background-image-width-xl",
    "type": "number",
  },
  "bi_w-xs": Object {
    "longLabel": "background-image-width-xs",
    "type": "number",
  },
  "bi_w-xxl": Object {
    "longLabel": "background-image-width-xxl",
    "type": "number",
  },
  "bi_w.u-g": Object {
    "default": "%",
    "longLabel": "background-image-width-unit-g",
    "type": "string",
  },
  "bi_w.u-l": Object {
    "longLabel": "background-image-width-unit-l",
    "type": "string",
  },
  "bi_w.u-m": Object {
    "longLabel": "background-image-width-unit-m",
    "type": "string",
  },
  "bi_w.u-s": Object {
    "longLabel": "background-image-width-unit-s",
    "type": "string",
  },
  "bi_w.u-xl": Object {
    "longLabel": "background-image-width-unit-xl",
    "type": "string",
  },
  "bi_w.u-xs": Object {
    "longLabel": "background-image-width-unit-xs",
    "type": "string",
  },
  "bi_w.u-xxl": Object {
    "longLabel": "background-image-width-unit-xxl",
    "type": "string",
  },
  "biw_h-g": Object {
    "default": 100,
    "longLabel": "background-image-wrapper-height-g",
    "type": "number",
  },
  "biw_h-l": Object {
    "longLabel": "background-image-wrapper-height-l",
    "type": "number",
  },
  "biw_h-m": Object {
    "longLabel": "background-image-wrapper-height-m",
    "type": "number",
  },
  "biw_h-s": Object {
    "longLabel": "background-image-wrapper-height-s",
    "type": "number",
  },
  "biw_h-xl": Object {
    "longLabel": "background-image-wrapper-height-xl",
    "type": "number",
  },
  "biw_h-xs": Object {
    "longLabel": "background-image-wrapper-height-xs",
    "type": "number",
  },
  "biw_h-xxl": Object {
    "longLabel": "background-image-wrapper-height-xxl",
    "type": "number",
  },
  "biw_h.u-g": Object {
    "default": "%",
    "longLabel": "background-image-wrapper-height-unit-g",
    "type": "string",
  },
  "biw_h.u-l": Object {
    "longLabel": "background-image-wrapper-height-unit-l",
    "type": "string",
  },
  "biw_h.u-m": Object {
    "longLabel": "background-image-wrapper-height-unit-m",
    "type": "string",
  },
  "biw_h.u-s": Object {
    "longLabel": "background-image-wrapper-height-unit-s",
    "type": "string",
  },
  "biw_h.u-xl": Object {
    "longLabel": "background-image-wrapper-height-unit-xl",
    "type": "string",
  },
  "biw_h.u-xs": Object {
    "longLabel": "background-image-wrapper-height-unit-xs",
    "type": "string",
  },
  "biw_h.u-xxl": Object {
    "longLabel": "background-image-wrapper-height-unit-xxl",
    "type": "string",
  },
  "biw_pos-g": Object {
    "default": "inherit",
    "longLabel": "background-image-wrapper-position-g",
    "type": "string",
  },
  "biw_pos-l": Object {
    "longLabel": "background-image-wrapper-position-l",
    "type": "string",
  },
  "biw_pos-m": Object {
    "longLabel": "background-image-wrapper-position-m",
    "type": "string",
  },
  "biw_pos-s": Object {
    "longLabel": "background-image-wrapper-position-s",
    "type": "string",
  },
  "biw_pos-xl": Object {
    "longLabel": "background-image-wrapper-position-xl",
    "type": "string",
  },
  "biw_pos-xs": Object {
    "longLabel": "background-image-wrapper-position-xs",
    "type": "string",
  },
  "biw_pos-xxl": Object {
    "longLabel": "background-image-wrapper-position-xxl",
    "type": "string",
  },
  "biw_pos.b-g": Object {
    "longLabel": "background-image-wrapper-position-bottom-g",
    "type": "string",
  },
  "biw_pos.b-l": Object {
    "longLabel": "background-image-wrapper-position-bottom-l",
    "type": "string",
  },
  "biw_pos.b-m": Object {
    "longLabel": "background-image-wrapper-position-bottom-m",
    "type": "string",
  },
  "biw_pos.b-s": Object {
    "longLabel": "background-image-wrapper-position-bottom-s",
    "type": "string",
  },
  "biw_pos.b-xl": Object {
    "longLabel": "background-image-wrapper-position-bottom-xl",
    "type": "string",
  },
  "biw_pos.b-xs": Object {
    "longLabel": "background-image-wrapper-position-bottom-xs",
    "type": "string",
  },
  "biw_pos.b-xxl": Object {
    "longLabel": "background-image-wrapper-position-bottom-xxl",
    "type": "string",
  },
  "biw_pos.b.u-g": Object {
    "default": "px",
    "longLabel": "background-image-wrapper-position-bottom-unit-g",
    "type": "string",
  },
  "biw_pos.b.u-l": Object {
    "longLabel": "background-image-wrapper-position-bottom-unit-l",
    "type": "string",
  },
  "biw_pos.b.u-m": Object {
    "longLabel": "background-image-wrapper-position-bottom-unit-m",
    "type": "string",
  },
  "biw_pos.b.u-s": Object {
    "longLabel": "background-image-wrapper-position-bottom-unit-s",
    "type": "string",
  },
  "biw_pos.b.u-xl": Object {
    "longLabel": "background-image-wrapper-position-bottom-unit-xl",
    "type": "string",
  },
  "biw_pos.b.u-xs": Object {
    "longLabel": "background-image-wrapper-position-bottom-unit-xs",
    "type": "string",
  },
  "biw_pos.b.u-xxl": Object {
    "longLabel": "background-image-wrapper-position-bottom-unit-xxl",
    "type": "string",
  },
  "biw_pos.l-g": Object {
    "longLabel": "background-image-wrapper-position-left-g",
    "type": "string",
  },
  "biw_pos.l-l": Object {
    "longLabel": "background-image-wrapper-position-left-l",
    "type": "string",
  },
  "biw_pos.l-m": Object {
    "longLabel": "background-image-wrapper-position-left-m",
    "type": "string",
  },
  "biw_pos.l-s": Object {
    "longLabel": "background-image-wrapper-position-left-s",
    "type": "string",
  },
  "biw_pos.l-xl": Object {
    "longLabel": "background-image-wrapper-position-left-xl",
    "type": "string",
  },
  "biw_pos.l-xs": Object {
    "longLabel": "background-image-wrapper-position-left-xs",
    "type": "string",
  },
  "biw_pos.l-xxl": Object {
    "longLabel": "background-image-wrapper-position-left-xxl",
    "type": "string",
  },
  "biw_pos.l.u-g": Object {
    "default": "px",
    "longLabel": "background-image-wrapper-position-left-unit-g",
    "type": "string",
  },
  "biw_pos.l.u-l": Object {
    "longLabel": "background-image-wrapper-position-left-unit-l",
    "type": "string",
  },
  "biw_pos.l.u-m": Object {
    "longLabel": "background-image-wrapper-position-left-unit-m",
    "type": "string",
  },
  "biw_pos.l.u-s": Object {
    "longLabel": "background-image-wrapper-position-left-unit-s",
    "type": "string",
  },
  "biw_pos.l.u-xl": Object {
    "longLabel": "background-image-wrapper-position-left-unit-xl",
    "type": "string",
  },
  "biw_pos.l.u-xs": Object {
    "longLabel": "background-image-wrapper-position-left-unit-xs",
    "type": "string",
  },
  "biw_pos.l.u-xxl": Object {
    "longLabel": "background-image-wrapper-position-left-unit-xxl",
    "type": "string",
  },
  "biw_pos.r-g": Object {
    "longLabel": "background-image-wrapper-position-right-g",
    "type": "string",
  },
  "biw_pos.r-l": Object {
    "longLabel": "background-image-wrapper-position-right-l",
    "type": "string",
  },
  "biw_pos.r-m": Object {
    "longLabel": "background-image-wrapper-position-right-m",
    "type": "string",
  },
  "biw_pos.r-s": Object {
    "longLabel": "background-image-wrapper-position-right-s",
    "type": "string",
  },
  "biw_pos.r-xl": Object {
    "longLabel": "background-image-wrapper-position-right-xl",
    "type": "string",
  },
  "biw_pos.r-xs": Object {
    "longLabel": "background-image-wrapper-position-right-xs",
    "type": "string",
  },
  "biw_pos.r-xxl": Object {
    "longLabel": "background-image-wrapper-position-right-xxl",
    "type": "string",
  },
  "biw_pos.r.u-g": Object {
    "default": "px",
    "longLabel": "background-image-wrapper-position-right-unit-g",
    "type": "string",
  },
  "biw_pos.r.u-l": Object {
    "longLabel": "background-image-wrapper-position-right-unit-l",
    "type": "string",
  },
  "biw_pos.r.u-m": Object {
    "longLabel": "background-image-wrapper-position-right-unit-m",
    "type": "string",
  },
  "biw_pos.r.u-s": Object {
    "longLabel": "background-image-wrapper-position-right-unit-s",
    "type": "string",
  },
  "biw_pos.r.u-xl": Object {
    "longLabel": "background-image-wrapper-position-right-unit-xl",
    "type": "string",
  },
  "biw_pos.r.u-xs": Object {
    "longLabel": "background-image-wrapper-position-right-unit-xs",
    "type": "string",
  },
  "biw_pos.r.u-xxl": Object {
    "longLabel": "background-image-wrapper-position-right-unit-xxl",
    "type": "string",
  },
  "biw_pos.sy-g": Object {
    "default": "all",
    "longLabel": "background-image-wrapper-position-sync-g",
    "type": "string",
  },
  "biw_pos.sy-l": Object {
    "longLabel": "background-image-wrapper-position-sync-l",
    "type": "string",
  },
  "biw_pos.sy-m": Object {
    "longLabel": "background-image-wrapper-position-sync-m",
    "type": "string",
  },
  "biw_pos.sy-s": Object {
    "longLabel": "background-image-wrapper-position-sync-s",
    "type": "string",
  },
  "biw_pos.sy-xl": Object {
    "longLabel": "background-image-wrapper-position-sync-xl",
    "type": "string",
  },
  "biw_pos.sy-xs": Object {
    "longLabel": "background-image-wrapper-position-sync-xs",
    "type": "string",
  },
  "biw_pos.sy-xxl": Object {
    "longLabel": "background-image-wrapper-position-sync-xxl",
    "type": "string",
  },
  "biw_pos.t-g": Object {
    "longLabel": "background-image-wrapper-position-top-g",
    "type": "string",
  },
  "biw_pos.t-l": Object {
    "longLabel": "background-image-wrapper-position-top-l",
    "type": "string",
  },
  "biw_pos.t-m": Object {
    "longLabel": "background-image-wrapper-position-top-m",
    "type": "string",
  },
  "biw_pos.t-s": Object {
    "longLabel": "background-image-wrapper-position-top-s",
    "type": "string",
  },
  "biw_pos.t-xl": Object {
    "longLabel": "background-image-wrapper-position-top-xl",
    "type": "string",
  },
  "biw_pos.t-xs": Object {
    "longLabel": "background-image-wrapper-position-top-xs",
    "type": "string",
  },
  "biw_pos.t-xxl": Object {
    "longLabel": "background-image-wrapper-position-top-xxl",
    "type": "string",
  },
  "biw_pos.t.u-g": Object {
    "default": "px",
    "longLabel": "background-image-wrapper-position-top-unit-g",
    "type": "string",
  },
  "biw_pos.t.u-l": Object {
    "longLabel": "background-image-wrapper-position-top-unit-l",
    "type": "string",
  },
  "biw_pos.t.u-m": Object {
    "longLabel": "background-image-wrapper-position-top-unit-m",
    "type": "string",
  },
  "biw_pos.t.u-s": Object {
    "longLabel": "background-image-wrapper-position-top-unit-s",
    "type": "string",
  },
  "biw_pos.t.u-xl": Object {
    "longLabel": "background-image-wrapper-position-top-unit-xl",
    "type": "string",
  },
  "biw_pos.t.u-xs": Object {
    "longLabel": "background-image-wrapper-position-top-unit-xs",
    "type": "string",
  },
  "biw_pos.t.u-xxl": Object {
    "longLabel": "background-image-wrapper-position-top-unit-xxl",
    "type": "string",
  },
  "biw_w-g": Object {
    "default": 100,
    "longLabel": "background-image-wrapper-width-g",
    "type": "number",
  },
  "biw_w-l": Object {
    "longLabel": "background-image-wrapper-width-l",
    "type": "number",
  },
  "biw_w-m": Object {
    "longLabel": "background-image-wrapper-width-m",
    "type": "number",
  },
  "biw_w-s": Object {
    "longLabel": "background-image-wrapper-width-s",
    "type": "number",
  },
  "biw_w-xl": Object {
    "longLabel": "background-image-wrapper-width-xl",
    "type": "number",
  },
  "biw_w-xs": Object {
    "longLabel": "background-image-wrapper-width-xs",
    "type": "number",
  },
  "biw_w-xxl": Object {
    "longLabel": "background-image-wrapper-width-xxl",
    "type": "number",
  },
  "biw_w.u-g": Object {
    "default": "%",
    "longLabel": "background-image-wrapper-width-unit-g",
    "type": "string",
  },
  "biw_w.u-l": Object {
    "longLabel": "background-image-wrapper-width-unit-l",
    "type": "string",
  },
  "biw_w.u-m": Object {
    "longLabel": "background-image-wrapper-width-unit-m",
    "type": "string",
  },
  "biw_w.u-s": Object {
    "longLabel": "background-image-wrapper-width-unit-s",
    "type": "string",
  },
  "biw_w.u-xl": Object {
    "longLabel": "background-image-wrapper-width-unit-xl",
    "type": "string",
  },
  "biw_w.u-xs": Object {
    "longLabel": "background-image-wrapper-width-unit-xs",
    "type": "string",
  },
  "biw_w.u-xxl": Object {
    "longLabel": "background-image-wrapper-width-unit-xxl",
    "type": "string",
  },
}
`;
